test/buy-tshirt.spec.ts

```diff
@@ -1,42 +1,64 @@
-import { $, browser } from 'protractor';
+import { browser } from 'protractor';
+import {
+  MenuContentPage,
+  SignInStepPage,
+  AddressStepPage,
+  SummaryStepPage,
+  ShippingStepPage,
+  PaymentStepPage,
+  OrderSummaryPage,
+  ProductAddedModalPage,
+  ProductListPage,
+  BankPaymentPage
+ } from '../src/page';
 
 describe('Buy a t-shirt', () => {
+  const menuContentPage: MenuContentPage = new MenuContentPage();
+  const signInStepPage: SignInStepPage = new SignInStepPage();
+  const addressStepPage: AddressStepPage = new AddressStepPage();
+  const summaryStepPage: SummaryStepPage = new SummaryStepPage();
+  const shippingStepPage: ShippingStepPage = new ShippingStepPage();
+  const paymentStepPage: PaymentStepPage = new PaymentStepPage();
+  const orderSummaryPage: OrderSummaryPage = new OrderSummaryPage();
+  const productAddedModalPage: ProductAddedModalPage = new ProductAddedModalPage();
+  const productListPage: ProductListPage = new ProductListPage();
+  const bankPaymentPage: BankPaymentPage = new BankPaymentPage();
   beforeEach(() => {
     jasmine.DEFAULT_TIMEOUT_INTERVAL = 120000;
   });
 
   it('then should be bought a t-shirt', async () => {
     await browser.waitForAngularEnabled(false);
     await browser.get('http://automationpractice.com/');
-    await(browser.sleep(10000));
-    await $('#block_top_menu > ul > li:nth-child(3) > a').click();
     await(browser.sleep(3000));
-    await $('#center_column a.button.ajax_add_to_cart_button.btn.btn-default').click();
+    await menuContentPage.goToTShirtMenu();
     await(browser.sleep(3000));
-    await $('[style*="display: block;"] .button-container > a').click();
+    await productListPage.selectAddToCart();
     await(browser.sleep(3000));
-    await $('.cart_navigation span').click();
+    await productAddedModalPage.goToProceedToCheckout();
+    await(browser.sleep(3000));
+    await summaryStepPage.goToProceedToCheckout();
     await(browser.sleep(3000));
 
-    await $('#email').sendKeys('aperdomobo@gmail.com');
-    await $('#passwd').sendKeys('WorkshopProtractor');
-    await $('#SubmitLogin > span').click();
+    await signInStepPage.sendEmailField('aperdomobo@gmail.com');
+    await signInStepPage.sendPasswdField('WorkshopProtractor');
+    await signInStepPage.goToLoginSubmitButton();
```

Comment:
This can be simplified a little bit more, what if you create a login or signIn method that receives both, pass and user, and it is responsible for populate the user and pass field, as well as click on the login button?

Reply:
Done